### PR TITLE
 Add TensorFlow ecosystem Spark and Hadoop jars to Maven deployment

### PR DESCRIPTION
**Pull request by skavulya**
Fixes https://github.com/tensorflow/ecosystem/issues/29 by adding TensorFlow ecosystem jars to public Maven repo.

@jsheu Not sure if this is the best way to do this. There were a couple of hacks I had to do since the poms in the TensorFlow ecosystem are decoupled from the TensorFlow java ones. I added a placeholder poms in tensorflow-hadoop and spark-tensorflow-connector to prevent errors during the initial mvn clean. I overwrite these poms with the ones from the ecosystem. I also added the deployment profiles to the ecosystem poms so we would need to manually change them if the TensorFlow java ones change.

An alternative approach I could use is to remove the TensorFlow ecosystem modules from the parent pom, and deploy the TensorFlow java jars first using `deploy_artifacts` then write a separate function to deploy the ecosystem jars. The advantage of this is that we could pass any changes to deployment settings via command-line and we wouldn't need placeholder poms for maven clean.

The previous approach I tried was dynamically editing the ecosystem poms using Python XML so that they would inherit from the parent pom but it made a number of hard-coded assumptions about the structure of the poms.

Please let me know what changes to make.

### CODE REVIEW

File: tensorflow/java/maven/pom.xml [hunk 1]
```diff
@@ -32,6 +32,8 @@
     <module>libtensorflow_jni_gpu</module>
     <module>tensorflow</module>
     <module>proto</module>
+    <module>tensorflow-hadoop</module>
```
Comment (asimshankar) [inline tensorflow/java/maven/pom.xml]: Should we make these names shorter, they are already in the `org.tensorflow` group, so:

- `org.tensorflow:hadoop` might be better than `org.tensorflow:tensorflow-hadoop`, and
- `org.tensorflow:spark-connector` might be better than `org.tensorflow:spark-tensorflow-connector'
?

File: tensorflow/java/maven/run_inside_container.sh [hunk 2]
```diff
@@ -240,14 +279,17 @@ cd "${DIR}"
 # Comment lines out appropriately if debugging/tinkering with the release
 # process.
 # gnupg2 is required for signing
-apt-get -qq update && apt-get -qqq install -y gnupg2
+apt-get -qq update && apt-get -qqq install -y gnupg2 && apt-get -qqq install -y git
```
Comment (asimshankar) [inline tensorflow/java/maven/run_inside_container.sh]: Will a single `apt-get install` command suffice? As in: `apt-get -qqq install -y gnupg2 git`

**Review by asimshankar — CHANGES_REQUESTED** [review]
Hmm...as you pointed out, the releases of these new artifacts seem decoupled with the TensorFlow release process (and versioning), so I'm not sure if this is the right place to add them. In particular, for the existing artifacts, we always upload a version of the code compiled from the corresponding release branch. For the new artifacts, we're uploading a version of the code from HEAD at the time the script was invoked.

Perhaps we need branches in the ecosystem repository?

@jhseu - What do you suggest?

**Comment by jhseu** [comment]
The code for this in ecosystem changes so infrequently that I'm not sure it's worth having versions (I think HEAD is fine). We can revisit it later.

**Comment by asimshankar** [comment]
@jhseu : Should we still create branches in the ecosystem repository so that we at least have a snapshot of the exact code used to produce a release (and use that branch here)?

**Comment by jhseu** [comment]
@asimshankar Yeah, that seems reasonable. I'll add it to the release instructions after this is merged.

@skavulya Mind modifying this CL to checkout the branch `"r${TF_VERSION}"?`

File: tensorflow/java/maven/pom.xml (hunk 1 above)
Comment (skavulya) [reply tensorflow/java/maven/pom.xml]: Great suggestion. I'll make the changes.

**Comment by skavulya** [comment]
@asimshankar @jhseu Thanks for the review. I'll make the changes you recommended.

**Comment by rmlarsen** [comment]
@skavulya Thanks for the update. @asimshankar another look?

**Comment by asimshankar** [comment]
@rmlarsen : I think @skavulya hasn't made the changes yet.

**Comment by rmlarsen** [comment]
@asimshankar Right, sorry about the ping :-)

**Comment by skavulya** [comment]
@asimshankar @jhseu I made the changes you requested. I was not able to test the full deployment for the latest changes due to errors downloading the nightly builds for the snapshot release, and missing libraries for the proto jar.

Please let me know if you would like me to make more changes.

**Comment by skavulya** [comment]
@asimshankar did you get a chance to look at the pull request? I added the `git checkout "r${TF_VERSION}"` suggested by @jhseu but you need to commit it out when testing because the branch does not yet exist in the TensorFlow ecosystem.

**Comment by jhseu** [comment]
Looks good to me. Asim, did you want to take a final pass?

**Review by asimshankar — DISMISSED** [review]
Could you also update the `.gitignore` file so that we don't track the downloaded src directories?

Otherwise looks good, thanks!

**Comment by skavulya** [comment]
Thanks @asimshankar @jhseu. I updated the .gitignore file

**Review by asimshankar — APPROVED** [review]
Thanks!

**Comment by skavulya** [comment]
Thanks @asimshankar

**Comment by jhseu** [comment]
Thanks for making this change, @skavulya!

**Comment by jsheu** [comment]
Please remove me, “jsheu”, from these notifications.  Somebody made a typo and confused me with “jhseu”.

**Pull request merged**